src/Native/Unix/System.Native/pal_io.cpp

```diff
@@ -364,6 +365,60 @@ extern "C" int32_t SystemNative_ReadDirR(DIR* dir, void* buffer, int32_t bufferS
     return 0;
 }
 
+#else
```

Comment:
Rather than have the if/else around the whole function, is it feasible to have it only around the parts that are different?  The signatures are the same. It looks like the preambles with argument validation are the same.  Both call ConvertDirent and return 0 at the end.  Etc.